src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random random = new Random();
+    private static final double MAX_RANDOM_VALUE = 10.;
+    private static final int AMOUNT_OF_FIGURES = 5;
+
+    public static Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        String color = colorSupplier.getRandomColor();
+
+        switch (random.nextInt(AMOUNT_OF_FIGURES)) {
+            case 0:
+                return new Square(color, random.nextDouble(MAX_RANDOM_VALUE));
+            case 1:
+                return new Circle(color, random.nextDouble(MAX_RANDOM_VALUE));
+            case 2:
+                return new IsoscelesTrapezoid(color,
+                        random.nextDouble(MAX_RANDOM_VALUE),random.nextDouble(MAX_RANDOM_VALUE),
+                        random.nextDouble(MAX_RANDOM_VALUE));
+            case 3:
+                return new RightTriangle(color,
+                        random.nextDouble(MAX_RANDOM_VALUE),
+                        random.nextDouble(MAX_RANDOM_VALUE));
+            default:
+                return new Rectangle(color,
+                        random.nextDouble(MAX_RANDOM_VALUE),
+                        random.nextDouble(MAX_RANDOM_VALUE));
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability.